.github/workflows/static_checks.yml

```diff
@@ -0,0 +1,34 @@
+name: Static checks
+on:
+  pull_request: {}
+  push:
+    paths-ignore:
+    - '*.md'
+    - '**/*.md'
+    branches:
+    - master
+jobs:
```

Comment:
What do you think about adding a `go_dependencies` job similar to the `linkerd/linkerd2` [repo](https://github.com/linkerd/linkerd2/blob/master/.github/workflows/static_checks.yml#L11)?

We'd have to add a script as well, but I think that's okay. The job and the script can be addressed in another PR.

Reply:
I don't think this is necessary, because in `linkerd/linkerd2` repo the `go_dependencies` job is making sure the Dockerfile is using the correct builder image.
In this repo, there is no such case to make sure the builder image.